acme/acme.go

```diff
@@ -331,6 +334,14 @@ func (a *ACME) CreateLocalConfig(tlsConfig *tls.Config, checkOnDemandDomain func
 func (a *ACME) getCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
 	domain := types.CanonicalDomain(clientHello.ServerName)
 	account := a.store.Get().(*Account)
+	//use regex to test for wildcard certs that might have been added into TLSConfig
+	for k := range a.TLSConfig.NameToCertificate {
+		selector := "^" + strings.Replace(k, "*.", ".*\\.?", -1) + "$"
```

Comment:
Ah. I see what you are saying. As is, it will serve the static wildcard.

Feel free to open a new PR (Or  a new issue with your proposed regex) to update the regex, and we can test it out. Im not against it. I think that if we are going to use ACME as a "fallback" when static certs are added, having them matched by a more accurate regex would be a good thing.